### ISSUE TITLE
possible to use this for sentiment controlled generation?

### ISSUE THREAD

**Issue by zichaow** [xxbidiao/plug-and-blend#1]
Hi, thanks for sharing the code! I was playing with the code and was wondering if I may be able to use it for sentiment controlled generation, in addition to topic controlled generation? Please correct me if I'm wrong, but I think this may be possible because the original GeDi paper demonstrated GeDi can be use for sentiment control, and some of GeDi components are used in this repo. 

In this repo, I have tried `Review Rating: 5.0` for positive sentiment and `Review Rating: 1.0` for negative sentiment as suggested in appendix C.3 in the original GeDi paper (more specifically, I set `topic={"Review Rating: 5.0": 1}` and `topic={"Review Rating: 1.0": 1}` for positive sentiment and negative sentiment, respectively), but I got the same generation output for both these cases (command: `text = gedi_skill_obj.generate_one_sentence(sentence=sentence, topic=topic)`). Did I do something wrong? Advice appreciated! Thanks in advance!

**Comment by xxbidiao** [comment]
Thank you for your thoughts! So sorry that I just realized you opened an issue here.

Yes, you are right! The framework should work on sentiment control "codes" too, given an appropriate control model, as the underlying process is not changed.

The reason why you see the generator giving the same sentence is a combination of two reasons: (1) The model does not know good enough how to extrapolate these "topics" (We also did experiments on zero-shot blending generation in-house, and observed similar problems as you described), and (2) You may need stronger control strength (pass `extra_args = {"disc_weight":60}` for 2x strength) before you can see a difference in generated sentences, as the control signal strength is not strong enough to override signals (logits) from the base model.

I'm currently looking into how these can be improved. Feel free to let me know if you have any more thoughts! :)
